cmd/main.go

```diff
@@ -50,6 +50,9 @@ func init() {
 // Main implements the main function of the caddy command.
 // Call this if Caddy is to be the main() of your program.
 func Main() {
+	if err := resetTerminalState(); err != nil {
```

Comment:
Doesn't this disable the terminal state altogether? Wouldn't that break the dependency -- in case it's being used?

Reply:
My understanding of their code is they're using it because they have interactive TUI. Running `go mod why` against the offending dependency shows:
```
2021-10-04 22:57:09  ~/projects/caddyserver/caddy git:(fix-4251) ✗ $ go mod why go.step.sm/cli-utils/ui
# go.step.sm/cli-utils/ui
github.com/caddyserver/caddy/v2/modules/caddypki
github.com/smallstep/certificates/authority
github.com/smallstep/certificates/kms
github.com/smallstep/certificates/kms/softkms
go.step.sm/cli-utils/ui
```
`softkms` uses the `go.step.sm/cli-utils/ui` package to prompt for password. The only password prompt Caddy has is on run of `caddy trust` which uses `github.com/FiloSottile/mkcert` which doesn't use `go.step.sm/cli-utils/ui` for the prompt.

Reply:
Ok, just making sure.